query-node/mappings/src/content/utils.ts

```diff
@@ -0,0 +1,382 @@
+      integrateAsset('avatarPhoto', result, asset) // changes `result` inline!
+      delete metaAsObject.avatarPhoto
+    }
+
+    // prepare language if needed
+    if (metaAsObject.language) {
+      result.language = await prepareLanguage(metaAsObject.language, db)
+    }
+
+    return result
+  }
+
+  // process channel category
+  if (type instanceof ChannelCategory) {
+    return ChannelCategoryMetadata.deserializeBinary(metadata).toObject()
+  }
+
+  // process video
+  if (type instanceof Video) {
+    const meta = VideoMetadata.deserializeBinary(metadata)
+    const metaAsObject = meta.toObject()
+    const result = metaAsObject as any as Video
+
+    // prepare video category if needed
+    if (metaAsObject.category !== undefined) {
+      result.category = await prepareVideoCategory(metaAsObject.category, db)
+    }
+
+    // prepare media meta information if needed
+    if (metaAsObject.mediaType) {
+      // prepare video file size if poosible
+      const videoSize = await extractVideoSize(assets, metaAsObject.video)
+
+      result.mediaMetadata = await prepareVideoMetadata(metaAsObject, videoSize)
+      delete metaAsObject.mediaType
+    }
+
+    // prepare license if needed
+    if (metaAsObject.license) {
+      result.license = await prepareLicense(metaAsObject.license)
+    }
+
+    // prepare thumbnail photo asset if needed
+    if (metaAsObject.thumbnailPhoto !== undefined) {
+      const asset = await extractAsset(metaAsObject.thumbnailPhoto, assets, db, blockNumber, actor)
+      integrateAsset('thumbnail', result, asset) // changes `result` inline!
+      delete metaAsObject.thumbnailPhoto
+    }
+
+    // prepare video asset if needed
+    if (metaAsObject.video !== undefined) {
+      const asset = await extractAsset(metaAsObject.video, assets, db, blockNumber, actor)
+      integrateAsset('media', result, asset) // changes `result` inline!
+      delete metaAsObject.video
+    }
+
+    // prepare language if needed
+    if (metaAsObject.language) {
+      result.language = await prepareLanguage(metaAsObject.language, db)
+    }
+
+    // prepare information about media published somewhere else before Joystream if needed.
+    if (metaAsObject.publishedBeforeJoystream) {
+      // this will change the `channel`!
+      handlePublishedBeforeJoystream(result, metaAsObject.publishedBeforeJoystream.date)
+    }
+
+    return result
+  }
+
+  // process video category
+  if (type instanceof VideoCategory) {
+    return VideoCategoryMetadata.deserializeBinary(metadata).toObject()
+  }
+
+  // this should never happen
+  throw `Not implemented type: ${type}`
+}
+
+export function convertContentActorToOwner(actor: ContentActor): typeof DataObjectOwner {
+  if (actor.isMember) {
+    const owner = new DataObjectOwnerMember()
+    owner.member = actor.asMember.toBn()
+
+    return owner
+  }
+
+  if (actor.isLead) {
+    // TODO: who will be owner?
+  }
+
+  if (actor.isCurator) {
+    // TODO: who will be owner?
```

Comment:
When a curator changes or creates a new channel/video, who will become the owner of newly uploaded assets?

Reply:
Mappings regarding the content owner have been improved. Now all the video/channel content is owned by the channel. 

Note: there still might be a different owner when the content is created via `runtime-modules/storage`'s `add_content` extrinsic.